### PR TITLE
feat(ld-select): use up-to-date css custom props

### PR DESCRIPTION
**Pull request by borisdiakur**
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/liquid/components/ld-select/ld-option-internal/ld-option-internal.css [hunk 1]
```diff
@@ -4,19 +4,19 @@
...
-  --ld-option-thm-col-bg-hover: var(--ld-col-rb1);
-  --ld-option-thm-col-bg-focus: var(--ld-col-rb1);
-  --ld-option-thm-col-bg-active: var(--ld-col-rb1);
+  --ld-option-thm-col: var(--ld-thm-primary);
+  --ld-option-thm-col-hover: var(--ld-thm-primary-hover);
+  --ld-option-thm-col-focus: var(--ld-thm-primary-focus);
+  --ld-option-thm-col-active: var(--ld-thm-primary-active);
+  --ld-option-thm-col-bg-hover: var(--ld-thm-primary-alpha-lowest);
+  --ld-option-thm-col-bg-focus: var(--ld-thm-primary-alpha-lowest);
+  --ld-option-thm-col-bg-active: var(--ld-thm-primary-alpha-lowest);
```
Comment (renet) [inline src/liquid/components/ld-select/ld-option-internal/ld-option-internal.css:19]: I discussed this with Nic, as the design uses`--ld-col-rb-010` (a new color added when Nic updated the colors in Figma), which is not a valid theme color. Still waiting for feedback, but I suggest to not change colors in this refectoring, if these changes do not reflect the design, unless Nic agrees. :)

**Review by renet — CHANGES_REQUESTED** [review]

File: src/liquid/components/ld-select/ld-option-internal/ld-option-internal.css (hunk 1 above)
Comment (renet) [reply src/liquid/components/ld-select/ld-option-internal/ld-option-internal.css:19]: PS: Shouldn't these changes result in updates on the screenshots? Are we missing an e2e tests that test these specific states, or am I missing something?

Comment (renet) [reply src/liquid/components/ld-select/ld-option-internal/ld-option-internal.css:19]:
```suggestion
  --ld-option-thm-col-bg-hover: var(--ld-thm-primary-highlight);
  --ld-option-thm-col-bg-focus: var(--ld-thm-primary-highlight);
  --ld-option-thm-col-bg-active: var(--ld-thm-primary-highlight);
```

Should now do the job without changing the color at all.

File: src/liquid/components/ld-select/ld-select.css [hunk 2]
```diff
@@ -270,35 +270,35 @@
...
     + .ld-select__icon,
     .ld-select__icon {
-      color: var(--ld-select-col-invalid);
+      color: var(--ld-select-invalid-col);
     }
 
     @media (hover: hover) {
       &:hover {
-        box-shadow: inset 0 0 0 0.1rem var(--ld-select-col-invalid);
+        box-shadow: inset 0 0 0 0.1rem var(--ld-select-invalid-col);
```
[the remark this answers is not in the document]
Comment (borisdiakur) [reply src/liquid/components/ld-select/ld-select.css:295]: This one is also used as a box-shadow color. I'll keep it as it is.

File: src/liquid/components/ld-select/ld-select.css [hunk 3]
```diff
@@ -270,35 +270,35 @@
...
     &:where(:not(.ld-select__btn-trigger--ghost)) {
-      box-shadow: inset 0 0 0 0.1rem var(--ld-select-col-invalid);
+      box-shadow: inset 0 0 0 0.1rem var(--ld-select-invalid-col);
     }
 
     + .ld-icon,
     + .ld-select__icon,
     .ld-select__icon {
-      color: var(--ld-select-col-invalid);
+      color: var(--ld-select-invalid-col);
```
[the remark this answers is not in the document]
Comment (borisdiakur) [reply src/liquid/components/ld-select/ld-select.css:290]: This one is also used as a box-shadow color. I'll keep it as it is.

File: src/liquid/components/ld-select/ld-select.css [hunk 4]
```diff
@@ -28,25 +28,25 @@
...
-  --ld-select-thm-col: var(--ld-thm-ocean-bg-primary);
-  --ld-select-thm-col-hover: var(--ld-col-rb-hover);
-  --ld-select-thm-col-focus: var(--ld-col-rb-focus);
-  --ld-select-thm-col-active: var(--ld-col-rb-active);
-  --ld-select-ghost-thm-col-bg-hover: var(--ld-col-rb-a01);
-  --ld-select-ghost-thm-col-bg-focus: var(--ld-col-rb-a02);
+  --ld-select-invalid-col: var(--ld-thm-error);
+  --ld-select-invalid-col-hover: var(--ld-thm-error-hover);
+  --ld-select-invalid-col-focus: var(--ld-thm-error-focus);
+  --ld-select-invalid-col-active: var(--ld-thm-error-active);
```
[the remark this answers is not in the document]
Comment (borisdiakur) [reply src/liquid/components/ld-select/ld-select.css:42]: I wouldn't over-engineer things here. If I really want to know what --ld-select-invalid-col is applied to, I make a quick search in the file. This is only component internal stuff, so I think it's fine as it is.

File: src/liquid/components/ld-select/ld-select.css [hunk 5]
```diff
@@ -28,25 +28,25 @@
...
+  --ld-select-invalid-disabled-bg-col: var(--ld-thm-error-disabled);
+  --ld-select-thm-col: var(--ld-thm-primary);
+  --ld-select-thm-col-hover: var(--ld-thm-primary-hover);
+  --ld-select-thm-col-focus: var(--ld-thm-primary-focus);
+  --ld-select-thm-col-active: var(--ld-thm-primary-active);
+  --ld-select-ghost-thm-bg-col-hover: var(--ld-thm-primary-alpha-lowest);
+  --ld-select-ghost-thm-bg-col-focus: var(--ld-thm-primary-alpha-low);
```
[the remark this answers is not in the document]
Comment (borisdiakur) [reply src/liquid/components/ld-select/ld-select.css:49]: You're deleting --ld-select-thm-col although it's in use?

[the remark this answers is not in the document]
Comment (renet) [reply src/liquid/components/ld-select/ld-select.css]: Oh, my bad. I just searched in the diff, not the whole file. 😅

[the remark this answers is not in the document]
Comment (renet) [reply src/liquid/components/ld-select/ld-select.css]: Updated the suggestion.

**Review by renet — APPROVED** [review]

**Pull request merged**